test/cloud_test.go

```diff
@@ -310,3 +314,48 @@ func TestNodePriceFromCSVWithCase(t *testing.T) {
 	}
 
 }
+
+func TestNodePriceFromCSVByClass(t *testing.T) {
+	n := &v1.Node{}
+	n.Spec.ProviderID = "fakeproviderid"
+	n.Labels = make(map[string]string)
+	n.Labels[v1.LabelZoneRegion] = "eastus2"
```

Comment:
Are we matching on region or region/Zone set?

Reply:
Region. v1.LabelZoneRegion is either topology.kubernetes.io/region or failure-domain.beta.kubernetes.io/region depending on k8s version. v1.LabelZoneFailureDomain would be the zone.